app/crawlers/base.py

```diff
@@ -0,0 +1,35 @@
+import atexit
+from contextlib import contextmanager
+from typing import List
+
+from selenium import webdriver
+
+from app.log import get_logger
+
+logger = get_logger("__name__")
+
+
+class Crawler:
+    def __init__(self):
+        self._driver = None
+        atexit.register(self.cleanup)
+
+    def cleanup(self):
+        logger.info(f"Exiting Crawler during run. Removing {self._driver}.")
+        if self._driver:
+            self._driver.quit()
```

Comment:
Tries to create as graceful a shutdown in selenium as possible (uses atexit hook if request died before completing)